examples/kafka_spark_demo/feature_repo/streaming_features_tutorial.ipynb

```diff
@@ -0,0 +1,877 @@
+    "\n",
+    "A typical pattern that we see is that data scientists use Feast to define, use, and share features, which greatly improves their productivity, while ML engineers use Feast to ensure that all the features are available for both training and serving. This notebook will first show how a data scientist can define and use stream features in Feast, and then how an ML engineer can use the Push API to ensure that those streaming features are available in production.\n",
+    "\n",
+    "All the necessary resources to run this notebook are in `kafka_spark_demo`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Setup the feature store"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Apply feature repository\n",
+    "We first run `feast apply` to register the data sources and features."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/Users/felixwang/feast/env/lib/python3.8/site-packages/pkg_resources/_vendor/packaging/specifiers.py:273: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release\n",
+      "  warnings.warn(\n",
+      "/Users/felixwang/feast/env/lib/python3.8/site-packages/pkg_resources/_vendor/packaging/version.py:111: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release\n",
+      "  warnings.warn(\n",
+      "/Users/felixwang/feast/sdk/python/feast/stream_feature_view.py:98: RuntimeWarning: Stream Feature Views are experimental features in alpha development. Some functionality may still be unstable so functionality can change in the future.\n",
+      "  warnings.warn(\n",
+      "Created entity \u001b[1m\u001b[32mdriver\u001b[0m\n",
+      "Created entity \u001b[1m\u001b[32mcustomer\u001b[0m\n",
+      "Created feature view \u001b[1m\u001b[32mcustomer_stats\u001b[0m\n",
+      "Created stream feature view \u001b[1m\u001b[32mdriver_hourly_stats_stream\u001b[0m\n",
+      "\n",
+      "Deploying infrastructure for \u001b[1m\u001b[32mcustomer_stats\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "!feast apply"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Spin up Kafka + Redis + Feast services\n",
+    "\n",
+    "We then use Docker Compose to spin up the services we need by running `docker-compose up` from the `kafka_spark_demo` directory.\n",
+    "* This leverages a script (in `kafka_demo/`) that creates a topic, reads from `feature_repo/data/driver_stats_stream.parquet`, generates newer timestamps, and emits them to the topic.\n",
+    "* This also deploys an instance of Redis.\n",
```

Comment:
Can you explain how these services are interacting with our demo notebook?

Reply:
added some clarification